src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/InProcess/EditorInProcess_Caret.cs

```diff
@@ -27,6 +27,33 @@ public async Task MoveCaretAsync(int position, CancellationToken cancellationTok
             view.Caret.MoveTo(point);
         }
 
+        public async Task WaitForCaretMoveAsync(CancellationToken cancellationToken)
```

Comment:
😨 this would be a big red flag. Strong likelihood of a flaky test if this is used.

Reply:
What's the alternative, and how might we identify similar problems in the future? The caret having moved to the desired location is the desired effect of InvokeGoToDefinitionAsync, and I don't think we've got any guarantees that everything will have executed by the time `Edit_GoToDefinition` has been fired, so we'll need some await mechanism.

Reply:
The ExecCommand call is definitely not blocking, and not is IOleCommandTarget which is the suggestion to use instead.

The old tests hardcoded the expected resulting position, but I don't like that, since it's hard to know and validate as a human.

I will probably just make this use a `WaitForCurrentLineText` approach.